nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -1931,12 +1960,18 @@ private boolean mayBeNullExpr(VisitorState state, ExpressionTree expr) {
         exprMayBeNull = false;
         break;
       case MEMBER_SELECT:
-        // A MemberSelectTree for a field dereference or method call cannot have a null symbol; see
-        // comments in matchMemberSelect()
-        exprMayBeNull = exprSymbol == null ? false : mayBeNullFieldAccess(state, expr, exprSymbol);
+        if (exprSymbol == null) {
```

Comment:
So, the change here is that the tree `t1` corresponding to module symbol `s1` can have a `t2=t1.getExpression()` such that `s2=ASTHelpers.getSymbol(t2)` is `null`. Correct? And that's the only case we are aware of or the only case up to JDK 17 (involving `MEMBER_SELECT` or `IDENTIFIER`, I mean, as opposed to some parenthesized expression)? 

So, with the added check in line 457, no such `t2` can flow into `matchDereference(...)`, which means no such `s2` ends up here. Right?

Reply:
Yes, that's exactly right.  The idea is if `matchDereference()` is called at line 461, and `tree.getExpression()` at that line is an identifier or member-select tree `t`, then `ASTHelpers.getSymbol(t)` cannot be null.  The check at line 457 stops the call at line 461 for member-select trees representing refs to modules, which is the case coming up in #533.  If NullAway crashes again in the future due to this type of issue, the error message should be more understandable.

Should I add any more docs / comments around this @lazaroclapp?

Reply:
I don't think so. A lot of the questions arise from me trying to understand what changed with this fix, but not what we are trying to assert here. I think the comments are fine for the code in its steady state.

Reply:
Ok, thanks!  I will go ahead and merge then